examples/ldc3_3d_HYMLS.py

```diff
@@ -0,0 +1,100 @@
+    postprocess_interface = Interface(interface.parameters, interface.nx_global, interface.ny_global, interface.nz_global,
+                                      interface.discretization.dim, interface.discretization.dof)
+
+    nx = postprocess_interface.discretization.nx
+    ny = postprocess_interface.discretization.ny
+    nz = postprocess_interface.discretization.nz
+
+    x_local = x.gather()
+    if x.Comm().MyPID() == 0:
+        data.append(mu, utils.compute_volume_averaged_kinetic_energy(x_local.array, postprocess_interface))
+
+        with open('ldc_bif_' + str(nx) + '_' + str(ny) + '_' + str(nz) + '.obj', 'wb') as f:
+            pickle.dump(data, f)
+
+
+def main():
+    for nx in [32]:
+        dim = 3
+        dof = 4
+        ny = nx
+        nz = nx
+
+        # Define the problem
+        parameters = {'Problem Type': 'Lid-driven cavity',
+                      # Problem parameters
+                      'Reynolds Number': 1,
+                      'Lid Velocity': 0,
+                      # Use a stretched grid
+                      'Grid Stretching Factor': 1.5,
+                      # Set a maximum step size ds
+                      'Maximum Step Size': 500}
+
+        # parameters['Bordered Solver'] = True
+
+        parameters['Solver'] = {}
+        parameters['Solver']['Iterative Solver'] = {}
+        parameters['Solver']['Iterative Solver']['Maximum Iterations'] = 400
+        parameters['Solver']['Iterative Solver']['Num Blocks'] = 100
+        parameters['Solver']['Iterative Solver']['Convergence Tolerance'] = 1e-7
+
+        parameters['Preconditioner'] = {}
+        parameters['Preconditioner']['Number of Levels'] = 2
+
+        comm = Epetra.PyComm()
+
+        data = Data()
+        parameters['Postprocess'] = lambda x, mu: postprocess(data, interface, x, mu)
+
+        interface = HYMLSInterface.Interface(comm, parameters, nx, ny, nz, dim, dof)
+        m = interface.map
+
+        continuation = Continuation(interface, parameters)
+
+        # Compute an initial guess
+        x0 = HYMLSInterface.Vector(m)
+        x0.PutScalar(0.0)
+        x0 = continuation.continuation(x0, 'Lid Velocity', 0, 1, 1)[0]
+
+        # Perform an initial continuation to Reynolds number 1500 without detecting bifurcation points
+        ds = 100
+        target = 1500
+        parameters['Newton Tolerance'] = 1e-4
+        x, mu = continuation.continuation(x0, 'Reynolds Number', 0, target, ds)
+
+        x_local = x.gather()
+        if comm.MyPID() == 0:
+            with open('Re_b_mu_' + str(nx) + '_' + str(ny) + '_' + str(nz) + '.npy', 'wb') as f:
+                numpy.save(f, mu)
+
+            with open('Re_b_x_' + str(nx) + '_' + str(ny) + '_' + str(nz) + '.npy', 'wb') as f:
+                numpy.save(f, x_local.array)
+
+
+if __name__ == '__main__':
+    # importing mpi4py makes sure MPI_Init is called before
+    # all the Epetra stuff, and MPI_Finalize after everthing 
+    # is deleted.
+    import mpi4py
```

Comment:
As I explained, this won't work. You have to put it before any other imports if you want it to be called first. Importing actually runs code in Python (including the MPI_Init in PyTrilinos). I also wanted to add this file only after I made sure I can generate all the necessary results with it.